vim-patch: :uniq Ex command

zeertzjq opened the pull request:
#### vim-patch:9.1.1477: no easy way to deduplicate text

Problem:  no easy way to deduplicate text
Solution: add the :uniq ex command
          (Hirohito Higashi)

closes: vim/vim#17538

https://github.com/vim/vim/commit/74f0a77bb9a170a25b03cacf640aa4e1b1d222b2

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>


#### vim-patch:9.1.1478: Unused assignment in ex_uniq()

Problem:  Unused assignment in ex_uniq() (after v9.1.1476)
Solution: Remove the assignment and the wrong comments above
          (zeertzjq).

closes: vim/vim#17596

https://github.com/vim/vim/commit/fc378a88d808ec1d44dd8582498c1b1b6af8e8d6


#### vim-patch:ca793e6: runtime(vim): Update base-syntax, match :uniq command

closes: vim/vim#17601

https://github.com/vim/vim/commit/ca793e60dbcf00a8d5658b0120965edb4b0f3fab

Co-authored-by: Doug Kearns <dougkearns@gmail.com>


#### vim-patch:9.1.1481: gcc complains about uninitialized variable

Problem:  gcc complains about uninitialized variable
          (Tony Mechelynck, after v9.1.1476)
Solution: initialize variable

https://github.com/vim/vim/commit/42d2c5e803cec6ec661217fa1ae630a2f7c463bc

Co-authored-by: Christian Brabandt <cb@256bit.org>


#### vim-patch:1c471ac: runtime(doc): update description of :uniq command

The examples mention the [u] flag, so at least the [u] flag should be
introduced before the examples.

Slightly reword the sentence about trailing/leading white space.

closes: vim/vim#17604

https://github.com/vim/vim/commit/1c471ac548cc883fdd5446c62a4b90cdcb6769ee

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>


#### vim-patch:26ebe21: runtime(doc): mismatch between the :uniq document's description and examples

closes: vim/vim#17612

https://github.com/vim/vim/commit/26ebe21caa93ae4c3d7ceb8e83e799b0dabc6271

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>

zeertzjq committed:
vim-patch:9.1.1477: no easy way to deduplicate text

Problem:  no easy way to deduplicate text
Solution: add the :uniq ex command
          (Hirohito Higashi)

closes: vim/vim#17538

https://github.com/vim/vim/commit/74f0a77bb9a170a25b03cacf640aa4e1b1d222b2

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>

zeertzjq committed:
vim-patch:9.1.1478: Unused assignment in ex_uniq()

Problem:  Unused assignment in ex_uniq() (after v9.1.1476)
Solution: Remove the assignment and the wrong comments above
          (zeertzjq).

closes: vim/vim#17596

https://github.com/vim/vim/commit/fc378a88d808ec1d44dd8582498c1b1b6af8e8d6

zeertzjq committed:
vim-patch:ca793e6: runtime(vim): Update base-syntax, match :uniq command

closes: vim/vim#17601

https://github.com/vim/vim/commit/ca793e60dbcf00a8d5658b0120965edb4b0f3fab

Co-authored-by: Doug Kearns <dougkearns@gmail.com>

lewis6991 commented:
Personally I don't think this is worth the +900 diffstat.

zeertzjq commented:
It's only actually +200

lewis6991 commented:
Admittedly I have attempted to run `:uniq` more than once assuming it existed. Though maybe something like `:! uniq` should also work if it doesn't already.

bfredl commented:
I tried things like `sort u /.*/` to skip all the text and keep the existing order exactly (because `:sort` promises to be stable) but it doesn't work, as `sort u` de-duplication only happens if the lines would form a sorted sequence when `uniq` was **not** used. An alternative might be to change this which will make the code of `:sort` slightly more complicated, but might still be simpler than a separate command.

h-east commented:
@justinmk 
> "how is this different than :sort u"

It's written there.

You seem to be reading the documentation and code with the intent of criticizing Vim, so you're not making a fair judgment.
This also influences the "90%" figure.  (It's not actually that redundant).

> except for the single difference being that text isn't sorted.

No.

> And it looks like ex_uniq and ex_sort don't share code, when they obviously could.

Of course, I think it's a good basic policy.
I had that option, but I didn't choose it this time. There's a clear reason for it.
(Just take a look at the 200 lines of code and run the commands and you'll see my choice...)

---

I don't want to read any more of your unfair opinions so I'll end it here. Thank you.

zeertzjq committed:
vim-patch:1c471ac: runtime(doc): update description of :uniq command

The examples mention the [u] flag, so at least the [u] flag should be
introduced before the examples.

Slightly reword the sentence about trailing/leading white space.

closes: vim/vim#17604

https://github.com/vim/vim/commit/1c471ac548cc883fdd5446c62a4b90cdcb6769ee

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>

zeertzjq committed:
vim-patch:9.1.1481: gcc complains about uninitialized variable

Problem:  gcc complains about uninitialized variable
          (Tony Mechelynck, after v9.1.1476)
Solution: initialize variable

https://github.com/vim/vim/commit/42d2c5e803cec6ec661217fa1ae630a2f7c463bc

Co-authored-by: Christian Brabandt <cb@256bit.org>

zeertzjq committed:
vim-patch:26ebe21: runtime(doc): mismatch between the :uniq document's description and examples

closes: vim/vim#17612

https://github.com/vim/vim/commit/26ebe21caa93ae4c3d7ceb8e83e799b0dabc6271

Co-authored-by: Hirohito Higashi <h.east.727@gmail.com>

justinmk commented:
Well, I won't block this. But we should be more selective about which vim features we take in the future. Based on https://github.com/vim/vim/commit/b90c2395b2c055aed38e0c5fd40c1841f43dab4b , Vim's goals are not Nvim's goals.
